### PR TITLE
Replace ringpop YARPC dispatcher with TChannel channel

### PR DESCRIPTION
**Pull request by alexshtin**
Ringpop doesn't require YARPC dispatcher but only channel from `tchannel` package.
This PR will help complete YARPC removal from the project (only few minor usages left).

### CODE REVIEW

File: common/service/config/ringpop/ringpop.go [hunk 1]
```diff
@@ -156,41 +150,14 @@ func (factory *RingpopFactory) getRingpop() (*membership.RingPop, error) {
...
 func (factory *RingpopFactory) broadcastAddressResolver() (string, error) {
 	// This initial piece is copied from ringpop-go/ringpop.go/channelAddressResolver
-	var ch *tcg.Channel
-	var err error
-	if ch, err = factory.getChannel(factory.dispatcher); err != nil {
-		return "", err
-	}
-
-	return membership.BuildBroadcastHostPort(ch.PeerInfo(), factory.config.BroadcastAddress)
+	return membership.BuildBroadcastHostPort(factory.channel.PeerInfo(), factory.config.BroadcastAddress)
```
Comment (alexshtin) [inline common/service/config/ringpop/ringpop.go]: @shawnhathaway, I am not sure if the comment above still relevant here.

File: common/service/config/rpc.go [hunk 2]
```diff
@@ -37,12 +36,11 @@ import (
 type RPCFactory struct {
 	config      *RPC
 	serviceName string
-	ch          *tchannel.ChannelTransport
 	logger      log.Logger
 
 	sync.Mutex
-	grpcListiner      net.Listener
```
Comment (alexshtin) [inline common/service/config/rpc.go]: image

File: common/service/config/ringpop/ringpop.go [hunk 3]
```diff
@@ -156,41 +150,14 @@ func (factory *RingpopFactory) getRingpop() (*membership.RingPop, error) {
...
 	}
 
 	return membership.NewRingPop(rp, factory.config.MaxJoinDuration, factory.logger), nil
 }
-
-func (factory *RingpopFactory) getChannel(
-	dispatcher *yarpc.Dispatcher,
-) (*tcg.Channel, error) {
-
-	t := dispatcher.Inbounds()[0].Transports()[0].(*tchannel.ChannelTransport)
```
Comment (alexshtin) [inline common/service/config/ringpop/ringpop.go]: This was tricky code.

File: common/rpc.go [hunk 4]
```diff
@@ -23,15 +23,15 @@ package common
 import (
 	"net"
 
-	"go.uber.org/yarpc"
```
Comment (shawnhathaway) [inline common/rpc.go]: 👍

File: common/service/config/ringpop/ringpop.go (hunk 1 above)
Comment (shawnhathaway) [reply common/service/config/ringpop/ringpop.go]: Yea comment isn't relevant anymore, feel free to pull out.

**Pull request merged**